src/shogun/mathematics/RandomNamespace.h

```diff
@@ -0,0 +1,122 @@
+#ifndef __RANDOM_NAMESPACE_H__
+#define __RANDOM_NAMESPACE_H__
+
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+
+#include <algorithm>
+#include <random>
+
+namespace shogun
+{
+	namespace random
+	{
+		/** Reorders the elements in [first,  last) randomly
+		 *
+		 * @param first an iterator to the first element in the range
+		 * @param last an iterator to the last element in the range
+		 */
+		template <typename RandomIt, typename PRNG>
+		static inline void shuffle(RandomIt first, RandomIt last, PRNG& prng)
+		{
+			std::shuffle(first, last, prng);
+		}
+
+		/** Reorders a container of elements randomly
+		 *
+		 * @param container the container holding the elements
+		 */
+		template <typename Container, typename PRNG>
+		static inline void shuffle(Container& container, PRNG& prng)
+		{
+			shuffle(container.begin(), container.end(), prng);
+		}
+
+		/**
+		 * Generate a positive signed random integer in the
+		 * range [0, PRNG::max() - PRNG::min()] if
+		 * signed(PRNG::max() - PRNG::min()) > 0, otherwise
+		 * the range is [0, std::limits<I>::max()]
+		 *
+		 * @return the random positive signed integer
+		 */
+		template <
+		    typename PRNG, typename I = typename std::make_signed<
+		                       typename PRNG::result_type>::type>
+		static inline I random_pos(PRNG& prng)
+		{
+			return (prng() - PRNG::min()) &
+			       ((typename PRNG::result_type(-1) << 1) >> 1);
+		}
+
+		/**
+		 * Get random
+		 * @return a float64_t random from [0,1] interval
+		 */
+		template <typename PRNG>
+		static inline float64_t random_close(PRNG& prng)
```

Comment:
why? i mean why not a UniformDistribution?

Reply:
I had the same thought. The previous implementation was this way, so I benchmarked uniform distribution against it, and uniform distribution is 3 much slower on my machine. Probably because it produces better results.

Reply:
i'm not so sure what you mean by being slower... which implementation? i mean it would be good that all this is actually factored out to class that gives you samples from uniform distribution in a given range... just like NormalDistribution

Reply:
my bad, it's much slower only without O3 flag, otherwise they're comparable.

Reply:
i've just done a git grep for this function and i dont see it anywhere being used in the main source... can we just drop all these functions, especially if not used